core/src/main/scala/org/apache/spark/scheduler/SchedulerBackend.scala

```diff
@@ -41,4 +41,11 @@ private[spark] trait SchedulerBackend {
    */
   def applicationId(): String = appId
 
+  /**
+   * Get an application ID associated with the job.
+   *
```

Comment:
when is this defined vs None? Is it as simple as "if the cluster manager provides it then it's defined, otherwise none"?